src/test/java/com/flab/offcoupon/service/coupon_issue/sync/RedissonLockCouponIssueTest.java

```diff
@@ -37,42 +32,20 @@ class RedissonLockCouponIssueTest {
 
     @Autowired
     private CouponRepository couponRepository;
+    private SetupUtils setupUtils = new SetupUtils();
 
     @BeforeEach
-    void setUp(){
-        Event event = new Event(
-                1L,
-                "바디케어",
-                "바디케어 전품목 이벤트",
-                LocalDate.now(),
-                LocalDate.now(),
-                "13:00:00",
-                "15:00:00",
-                LocalDateTime.now(),
-                LocalDateTime.now());
-        eventRepository.save(event);
-
-        Coupon coupon = new Coupon(
-                1L,
-                1L,
-                DiscountType.PERCENT,
-                50L,
-                null,
-                CouponType.FIRST_COME_FIRST_SERVED,
-                500L,
-                0L,
-                LocalDateTime.now().plusMonths(1L),
-                LocalDateTime.now().plusMonths(2L),
-                LocalDateTime.now(),
-                LocalDateTime.now());
-        couponRepository.save(coupon);
+    void setUp() {
+        setupUtils.setUpEventAndCoupon(eventRepository, couponRepository);
```

Comment:
이렇게 `set~` 메서드를 통해 무조건 초기화를 하고 사용해야 하는 경우, 1. 객체생성 -> 2. 초기화로 과정이 두개의 단계를 가지게 되고, 경우에 따라서는 과정이 누락되거나 초기화 전 객체의 사용이 발생될 수 있습니다.

그런 경우가 발생하지 않도록 아예 객체 생성시 꼭 필요한 상태를 생성자로 주입해주면 객체 생성부터 세팅까지를 한단계로 줄일 수 있으니 생성자를 통해 상태를 주입해주세요~

각 테스트마다 독립된 셋업이라면 이렇게 만들어줄 수 있습니다.


```suggestion
        setupUtils  =new SetupUtils(eventRepository, couponRepository);
```

Reply:
초기화 메소드 내부에서 생성자 주입하는게 휴먼 에러를 방지하고 훨씬 유지보수에 쉬운거군요 감사합니당~